File: prototype/rails/src/test/domain/reports/activity_totals_test.rb
require "test_helper"

module Domain
  module Reports
    class ActivityTotalsTest < ActiveSupport::TestCase
      test "it reads the count of each event the report shows" do
        totals = ActivityTotals.from({ "view" => 12, "favorite" => 3, "cart_add" => 2 })

        assert_equal 12, totals.views
        assert_equal 3, totals.favorites
        assert_equal 2, totals.cart_adds
      end

      test "an event that has not happened counts zero" do
        totals = ActivityTotals.from({ "view" => 12 })

        assert_equal 0, totals.favorites
        assert_equal 0, totals.cart_adds
      end

      test "it ignores events no report shows" do
        assert_equal 0, ActivityTotals.from({ "unfavorite" => 5 }).total
      end

      test "it sums the three event kinds" do
        assert_equal 17, ActivityTotals.from({ "view" => 12, "favorite" => 3, "cart_add" => 2 }).total
      end

      test "a listing nobody has seen totals zero" do
        assert_equal 0, ActivityTotals.from({}).total
      end
    end
  end
end

File: prototype/node/src/app/core/reports/activity-totals.test.ts
import { test } from 'node:test'
import assert from 'node:assert/strict'
import { activityTotals } from './activity-totals.ts'

test('it reads the count of each event the report shows', () => {
  const totals = activityTotals({ view: 12, favorite: 3, cart_add: 2 })

  assert.equal(totals.views, 12)
  assert.equal(totals.favorites, 3)
  assert.equal(totals.cartAdds, 2)
})

test('an event that has not happened counts zero', () => {
  const totals = activityTotals({ view: 12 })

  assert.equal(totals.favorites, 0)
  assert.equal(totals.cartAdds, 0)
})
